### ISSUE TITLE
Add get namespace to itsyouonline.Client

### ISSUE THREAD

**Issue by iwanbk** [zero-os/0-stor#504]
With basic info:
- permission
- creation date

**Comment by GlenDC** [comment]
I haven't checked the itsyou.online REST API, but I suppose if it supports it we can indeed easily support it as well, and add it to the Daemon's Namespace Service and as zerostor's namespace sub commands as well.

Can you please specify however why you need this feature. Our dev resources are limited and spread at the moment, so for the initial launch I would like to keep the amount of features we have to implement to a minimum. Therefore I would like to read a proper motivation, backed by evidence that this feature is absolutely required. Otherwise I'll move it to roadmap instead.

**Comment by iwanbk** [comment]
> Can you please specify however why you need this feature

It is for minio integration, i don't think that it is really important, you can move it to `roadmap` or assign it to me.

**Comment by GlenDC** [comment]
We can get the namespace, but we won't get all info we get from an organization (a namespace is a set made out of an organization and some sub (permission) organizations) that you want. Creation date doesn't seem to be something they expose/store. Here is what we can get from an organization:

```raml
Organization:
    properties:
      globalid:
        type: string
        minLength: 3
        maxLength: 150
        pattern: ^[a-z\d\-_\s]{3,150}$
      publicKeys:
        type: string[]
        maxItems: 20
      dns:
        type: string[]
        maxItems: 100
        description: List of organization DNS.
      owners:
        type: string[]
        maxItems: 20
        description: List of users `usernames`.
      members:
        type: string[]
        maxItems: 2000
        description: List of users who are member in this organization.
      orgowners:
        type: string[]
        description: List of other organizations who act as an owner in this organization.
      orgmembers:
        type: string[]
        description: List of other organizations who act as a member in this organization.
      includes:
        type: string[]
        maxItems: 100
        description: globalId of sub organizations
      requiredscopes:
        type: RequiredScope[]
        maxItems: 20
        description: List of scopes that are required to join this organization
      includesuborgsof:
        type: string[]
        description: List of orgowners and orgmembers who's children should be included in the organizations membershi or ownership hierarchy
```
>  full spec: https://itsyou.online/apidocumentation/raml/itsyouonline.raml

We already know how to get the permissions, so that will be possible. That's just a matter of checking if a user is part of all those sub organizations.

The creation date is for now however not possible AFAIK. This probably can be seen as extra namespace metadata, specific to 0-stor, just as need to be able to store a public read-only property for a namespace (see #352).

The trick is to find a good (de)centralized place where we can store this. It was originally proposed to simply store this in all the datastor shards, but this is a very hacky and error-prone way. It's really not something I would go for. I think the ideal solution would be to have it stored via some namespace service that runs as a separate (remote) process. Ideally that service would store it on a blockchain, but for now it could simply be a (distributed) database. Nothing is designed for this yet however, so if we want this for the TF launch (which we probably do), we'll have to come up with a design fast.

**Comment by iwanbk** [comment]
> so if we want this for the TF launch (which we probably do), we'll have to come up with a design fast.

If IYO really doesn't have the creation date, i'd rather move it to roadmap.

**Comment by GlenDC** [comment]
Yeah we can do, for now I won't do it.

As I said, we do really need #352 (this is required for the TF launch, right @zaibon?). And once we figure out how to store that `public read-only` metadata property for a namespace, it won't take any extra effort to add this creation date property.

We'll see, I'll already note that if we can't get around to do so, that I won't have to be afraid to move it to the roadmap.

**Comment by GlenDC** [comment]
Yeah we can do, for now I won't do it.

As I said, we do really need #352 (this is required for the TF launch, right @zaibon?). And once we figure out how to store that `public read-only` metadata property for a namespace, it won't take any extra effort to add this creation date property.

We'll see, I'll already note that if we can't get around to do so, that I won't have to be afraid to move it to the roadmap.

**Comment by lafont-e** [comment]
As I see it:
There are 2 functions on 0-stor.Client that can help with this
* ListNamespaces() That returns the tree of the organization defined in the conf of the Client, this function has all Namespaces, usually 1 but can represent a multitude when listing a big tree.
* GetPermisions() , given a namespace (full path) returns the permissions of such namespace

This two functions may solve your need, or if I'm missing something, please let me know

**Comment by GlenDC** [comment]
Getting permissions is not the issue.

This issue is however blocked on getting the creation date, as that is the thing that isn't available yet.

**Comment by zaibon** [comment]
The more I think about it the more I think namespace should be something that is part of the metadata.
At the end the 0-stor is nothing more then just exposing storage capacity, all the logic already lies in the client.
And after writting a file to a 0-stor cluster, the client needs to store the metadata somewhere.
So while doing so it can easily organize the metadata per namespace, since it is vital for the client to make sure the metadta is always correct, otherwise it can't retreive anything from 0-stor.
Having that in mind metadata store seems to be the correct location to store namespace information for a specific user.

**Comment by zaibon** [comment]
need to redefine this, cause after #515 this wont require IYO anymore
